### PR TITLE
add sdk permission check to StatisticProtocolServer

### PR DESCRIPTION
**Pull request by cyjseagull**

### CODE REVIEW

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -73,4 +73,4 @@ target_include_directories(test-fisco-bcos PRIVATE ..)
 target_include_directories(test-fisco-bcos PRIVATE ../utils)
 target_include_directories(test-fisco-bcos PRIVATE ${ROCKSDB_INCLUDE_DIR})
 target_link_libraries(test-fisco-bcos Boost::UnitTestFramework)
-target_link_libraries(test-fisco-bcos initializer)
+target_link_libraries(test-fisco-bcos initializer channelserver)
```
Comment (bxq2011hust) [inline test/CMakeLists.txt:76]: why this? I think this should not add

File: librpc/Rpc.cpp [hunk 2]
```diff
@@ -57,7 +57,8 @@ std::map<int, std::string> dev::rpc::RPCMsg{{RPCExceptionType::Success, "Success
     {RPCExceptionType::InvalidRequest,
         "Don't send request to this node who doesn't belong to the group"},
     {RPCExceptionType::IncompleteInitialization, "RPC module initialization is incomplete."},
-    {RPCExceptionType::OverQPSLimit, "Over QPS limit"}};
+    {RPCExceptionType::OverQPSLimit, "Over QPS limit"},
+    {RPCExceptionType::PermissionDenied, " PermissionDenied for not in the allowlist"}};
```
Comment (bxq2011hust) [inline librpc/Rpc.cpp:61]: The SDK is not allowed to access this group. maybe more clear

File: libledger/Ledger.cpp [hunk 3]
```diff
@@ -109,6 +113,25 @@ bool Ledger::initLedger(std::shared_ptr<LedgerParamInterface> _ledgerParams)
             initEventLogFilterManager());
 }
 
+void Ledger::initSDKAllowList()
```
Comment (bxq2011hust) [inline libledger/Ledger.cpp:116]: not support auto-reload?

File: libchannelserver/ChannelRPCServer.cpp [hunk 4]
```diff
@@ -1259,3 +1278,38 @@ std::vector<dev::channel::ChannelSession::Ptr> ChannelRPCServer::getSessionByTop
 
     return activedSessions;
 }
+
+void ChannelRPCServer::registerSDKAllowListByGroupId(
+    dev::GROUP_ID const& _groupId, dev::PeerWhitelist::Ptr _allowList)
+{
+    if (!getSDKAllowListByGroupId(_groupId))
```
Comment (bxq2011hust) [inline libchannelserver/ChannelRPCServer.cpp:1285]: this means not auto-reload?

File: libledger/LedgerParam.cpp [hunk 5]
```diff
@@ -689,5 +690,36 @@ void LedgerParam::initFlowControlConfig(boost::property_tree::ptree const& _pt)
                                  mutableFlowControlParam().outGoingBandwidthLimit);
 }
 
+void LedgerParam::initNodeListBySection(dev::h512s& _nodeList,
```
Comment (bxq2011hust) [inline libledger/LedgerParam.cpp:693]: confused about this function, not used?

File: libledger/LedgerParam.cpp [hunk 6]
```diff
@@ -689,5 +690,36 @@ void LedgerParam::initFlowControlConfig(boost::property_tree::ptree const& _pt)
...
+        boost::to_lower(data);
+        LedgerParam_LOG(INFO) << LOG_BADGE("initNodeListBySection")
+                              << LOG_KV("sectionName", _sectionName) << LOG_KV("it.first", data);
+        _nodeList.push_back(dev::h512(data));
+    }
+}
+
+void LedgerParam::initPermissionParam(boost::property_tree::ptree const& _pt)
+{
+    initNodeListBySection(m_permissionParam.sdkAllowList, _pt, "sdk_allowlist", "cal.");
```
Comment (bxq2011hust) [inline libledger/LedgerParam.cpp:719]: cal -> public_key?
what does cal mean?

File: libledger/LedgerParam.h [hunk 7]
```diff
@@ -182,6 +182,11 @@ struct FlowControlParam
     int64_t maxBurstReqNum;
 };
 
+struct PermissionParam
+{
+    dev::h512s sdkAllowList;
```
Comment (bxq2011hust) [inline libledger/LedgerParam.h:187]: maybe just a dev::h512s is more clear?

File: test/CMakeLists.txt (hunk 1 above)
Comment (cyjseagull) [reply test/CMakeLists.txt:76]: The link will be failed of test-fisco-bcos when compiling on linux if not add the `channelserver` here.

File: librpc/Rpc.cpp (hunk 2 above)
Comment (cyjseagull) [reply librpc/Rpc.cpp:61]: modified.

File: libledger/LedgerParam.cpp (hunk 6 above)
Comment (cyjseagull) [reply libledger/LedgerParam.cpp:724]: Done.

File: libledger/LedgerParam.cpp (hunk 5 above)
Comment (cyjseagull) [reply libledger/LedgerParam.cpp:698]: I think this function maybe used multi-time in the future versions.

Comment (bxq2011hust) [reply libledger/LedgerParam.cpp:682]: parseListOfSection？

File: test/CMakeLists.txt (hunk 1 above)
Comment (bxq2011hust) [reply test/CMakeLists.txt:76]: ok

File: libledger/LedgerParam.cpp (hunk 5 above)
Comment (cyjseagull) [reply libledger/LedgerParam.cpp]: Done.

**Pull request merged**